lib/stat_tracker.rb

```diff
@@ -1,13 +1,90 @@
 class StatTracker
-  attr_reader :locations
+  attr_reader :files,
+              :game_stats,
+              :team_stats,
+              :league_stats
 
-  def initialize(locations)
-    @locations = locations
-    #@game_info = file_converter.new(locations[:games]).game_info_converter
+  def initialize(files)
+    @files = files
+    @game_stats = compile_game_stats
+    @team_stats = compile_team_stats
+    @league_stats = compile_league_stats
   end
 
-  def self.from_csv(locations)
-    new(locations)
+  def self.from_csv(files)
+    new(files)
+  end
+
+  # StatCompiler superclass
+  # ======================================
+  def convert_files
+    FileConverter.new(files)
+  end
+
+  def compile_game_stats
+    games = convert_files.games
+    GameStats.new(games)
+  end
+
+  def compile_team_stats
+    teams = convert_files.teams
+    TeamStats.new(teams)
+  end
+
+  def compile_league_stats
+    games = convert_files.games
+    teams = convert_files.teams
+    LeagueStats.new(games, teams)
+  end
+#Compile season stats, game teams stats
+# ======================================
```

Comment:
Think it's fine for these all to be left together, or I see potential for some of these to move into a "seasons" class methods. I also see a scenario where "game_stats" could basically be renamed "seasons". A season is basically a grouping of games and some methods could be organized there. But this is fine for now, that could be discussed at a later time.